### ISSUE TITLE
Sapping is possible even without sufficient energy

### ISSUE THREAD

**Issue by LeFiz** [Lux-AI-Challenge/Lux-Design-S3#13]
Units can sap even when not holding enough energy (they kill themselves then).

It seems to be caused by

`team_unit_sapped = units_mask[t] & team_sap_action_mask & (current_energy[t, 0] >= params.unit_sap_cost) & (jnp.max(jnp.abs(team_sap_action_deltas), axis=-1) <= params.unit_sap_range) `

Where `(current_energy[t, 0] >= params.unit_sap_cost)` should be `(current_energy[t, :, 0] >= params.unit_sap_cost)` probably (need more testing though).
